File: tourneyhub/src/dto/user/UserDto.ts
import { ICountryDto } from '../ICountryDto';
import { TournamentRoleDto } from '../tournamentRole/TournamentRoleDto';
import { TournamentStatsDto } from '../TournamentStatsDto';

export interface UserDto {
    id: number,
    playerId: number,
    name: string,
    rank: number,
    country: ICountryDto,
    discordUsername: string,
    timezone: number,
    avatar: string,
    roles: TournamentRoleDto[],
    stats: TournamentStatsDto[]
}

File: backend/src/main/java/com/tourneyhub/backend/dto/UserDto.java
package com.tourneyhub.backend.dto;

import com.tourneyhub.backend.dto.tournamentRole.TournamentRoleDto;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class UserDto {

    private Integer playerId;

    private String name;

    private Integer rank;

    private String discordUsername;

    private Integer timezone;

    private String avatar;

    private CountryDto country;

    private List<TournamentRoleDto> roles;

    private List<TournamentStatsDto> stats;
}
